web_src/js/features/gitgraph.js

```diff
@@ -46,6 +46,67 @@ export default async function initGitGraph() {
       window.history.replaceState({}, '', window.location.pathname);
     }
   });
+  const url = new URL(window.location);
+  const params = url.searchParams;
+  const updateGraph = () => {
+    const queryString = params.toString();
+    let ajaxUrl = url.toString();
+    if (queryString) {
+      url.search = `?${queryString}`;
+      window.history.replaceState({}, '', `?${queryString}`);
+      ajaxUrl += '&div-only=true';
+    } else {
+      url.search = '';
+      window.history.replaceState({}, '', window.location.pathname);
+      ajaxUrl += '?div-only=true';
+    }
+    $('#pagination').html('');
+    $('#rel-container').addClass('hide');
+    $('#rev-container').addClass('hide');
+    $('#loading-indicator').removeClass('hide');
+
+    $.ajax(ajaxUrl).then((div) => {
+      $('#pagination').html($($.parseHTML(div)).find('#pagination').html());
+      $('#rel-container').html($($.parseHTML(div)).find('#rel-container').html());
+      $('#rev-container').html($($.parseHTML(div)).find('#rev-container').html());
+      $('#loading-indicator').addClass('hide');
+      $('#rel-container').removeClass('hide');
+      $('#rev-container').removeClass('hide');
+    });
+  };
+  $('#flow-hide-pr-refs').on('click', () => {
+    let hidePRRefs = true;
+    if (params.has('hide-pr-refs')) {
+      hidePRRefs = params.get('hide-pr-refs') === 'false';
+    }
+    if (hidePRRefs) {
+      $('#flow-hide-pr-refs').addClass('active');
+      params.set('hide-pr-refs', hidePRRefs);
+    } else {
+      $('#flow-hide-pr-refs').removeClass('active');
+      $('#flow-hide-pr-refs').blur();
+      params.delete('hide-pr-refs');
+    }
+    updateGraph();
+  });
+  $('#flow-select-refs-dropdown').dropdown('set selected', params.getAll('branch'));
+  $('#flow-select-refs-dropdown').dropdown({
+    clearable: true,
+    onRemove(_text) {
```

Comment:
Why the underscore? It usually indicates a unused variable name but this one is used.